frontend/js/src/utils/SearchTrackOrMBID.tsx

```diff
@@ -0,0 +1,192 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { throttle, throttle as _throttle } from "lodash";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import GlobalAppContext from "./GlobalAppContext";
+
+const RECORDING_MBID_REGEXP = /^(https?:\/\/(?:beta\.)?musicbrainz\.org\/recording\/)?([0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12})/i;
+type SearchTrackOrMBIDProps = {
+  onSelectRecording: (selectedRecordingMetadata: TrackMetadata) => void;
+  newAlert: (
+    alertType: AlertType,
+    title: string,
+    message: string | JSX.Element
+  ) => void;
+};
+
+export default function SearchTrackOrMBID({
+  onSelectRecording,
+  newAlert,
+}: SearchTrackOrMBIDProps) {
+  const { APIService } = useContext(GlobalAppContext);
+  const { lookupMBRecording } = APIService;
+  const [inputValue, setInputValue] = useState("");
+  const [searchResults, setSearchResults] = useState<Array<ACRMSearchResult>>(
+    []
+  );
+
+  const handleError = React.useCallback(
+    (error: string | Error, title?: string): void => {
+      if (!error) {
+        return;
+      }
+      newAlert(
+        "danger",
+        title || "Error",
+        typeof error === "object" ? error.message : error
+      );
+    },
+    [newAlert]
+  );
+
+  const throttledSearchTrack = useMemo(
+    () =>
+      throttle(
+        async (searchString: string) => {
+          try {
+            const response = await fetch(
+              "https://labs.api.listenbrainz.org/recording-search/json",
+              {
+                method: "POST",
+                body: JSON.stringify([{ query: searchString }]),
+                headers: {
+                  "Content-type": "application/json; charset=UTF-8",
+                },
+              }
+            );
+
+            const parsedResponse = await response.json();
+            setSearchResults(parsedResponse);
+          } catch (error) {
+            handleError(error);
+          }
+        },
+        800,
+        { leading: false, trailing: true }
+      ),
+    [handleError, setSearchResults]
+  );
+
+  const throttledHandleValidMBID = useMemo(
+    () =>
+      throttle(
+        async (input: string) => {
+          const newRecordingMBID = RECORDING_MBID_REGEXP.exec(
+            input
+          )![2].toLowerCase();
+
+          try {
+            const lookupResult: MusicBrainzRecordingWithReleases = (await lookupMBRecording(
```

Comment:
Alternatively,  you can use [the LB api](https://api.listenbrainz.org/1/metadata/recording/?recording_mbids=78194fbe-3bcd-4afe-b4a6-7a97711c325f&inc=artist%20release%20tag). The one benefit is that it will give you the canonical release metadata.

Reply:
The benefit is that it will be consistent with labs api which also returns the canonical recording.